### PR TITLE
feat: extract strata dimensions from mira model

### PR DESCRIPTION
**Pull request by shawnyama**
# Description

Mira matrices appear based on their base model variables.
We are temporarily showing both the normal and the stratified configurations for when we view a stratified model:

https://github.com/DARPA-ASKEM/Terarium/assets/28237258/c4e137d8-90f8-4053-9ef3-419ea2af2939

To make both configurations sync with each other I wrote `syncConfigs` in `tera-model.vue`. I then emit `syncConfigs` to the model configuration components. I don't think there is a great way to sync the two sibling components but it should do the temporary job.

### CODE REVIEW

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/Model.java
Comment (dvince2) [inline packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/Model.java:45]: Did `ModelMetadata` change? We should use the POJO where possible

File: packages/client/hmi-client/src/components/models/tera-stratified-value-matrix.vue
Comment (dvince2) [inline packages/client/hmi-client/src/components/models/tera-stratified-value-matrix.vue:13]: I absolutely could be wrong, but, in this block of code you have `v-for"i in matrix[0].length+1`.  Wouldn't that just be scoped to elements inside that `<th` tag and whats inside of it (`<Dropdown...`)?  Down below in another `<tr><th>` tree you reference i which seems wrong or if its not, I'd love a quick primer on how that works

**Review by dvince2 — CHANGES_REQUESTED** [review]

**Review by dvince2 — APPROVED** [review]
Items resolved over slack. looks good!

File: packages/client/hmi-client/src/model-representation/petrinet/petrinet-service.ts
Comment (mwdchang) [inline packages/client/hmi-client/src/model-representation/petrinet/petrinet-service.ts:601]: This function is now returning mixed types

**Comment by mwdchang** [comment]
Putting this back into draft - we can't merge this because it hardwires everything as a static mira-model.

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/modelparts/ModelMetadata.java
Comment (dvince2) [inline packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/modelparts/ModelMetadata.java:18]: If we're marking this as deprecated would you mind throwing a javadoc with a note of what people should be using now?

**Review by dvince2 — COMMENTED** [review]
Back end looks good - I'll let others weigh in on the front end.  Just make a note of what should be used with `ModelMetadata` being deprecated

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/modelparts/ModelMetadata.java
Comment (mwdchang) [reply packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/modelparts/ModelMetadata.java:18]: I think that came from me, ModelMetadata should be removed.

File: packages/client/hmi-client/src/components/models/tera-stratified-value-matrix.vue
Comment (YohannParis) [inline packages/client/hmi-client/src/components/models/tera-stratified-value-matrix.vue:209]: Why not ?

```JS
matrix.value.forEach(m => editableCellStates.value.push(Array(m.length).fill(false)));
```

**Review by YohannParis — APPROVED** [review]

**Pull request merged**
